### Librerias

In [1]:
import numpy as np
import pandas as pd
import plotly as py
import streamlit as st

In [14]:
# Cargamos el dataset y mostramos los primeros 10 registros

talks_df = pd.read_csv('raw_data/ted_talks.csv')
talks_df.head(10)

title  \
0      Climate action needs new frontline leadership   
1        The dark history of the overthrow of Hawaii   
2     How play can spark new ideas for your business   
3  Why is China appointing judges to combat clima...   
4     Cement's carbon problem — and 2 ways to fix it   
5  The tragedy of air pollution — and an urgent d...   
6                     The myth of Narcissus and Echo   
7         You deserve the right to repair your stuff   
8  What nature can teach us about sustainable bus...   
9     The origins of blackface and Black stereotypes   

                       author            date   views  likes  \
0        Ozawa Bineshi Albert   December 2021  404000  12000   
1               Sydney Iaukea   February 2022  214000   6400   
2               Martin Reeves  September 2021  412000  12000   
3           James K. Thornton    October 2021  427000  12000   
4             Mahendra Singhi    October 2021    2400     72   
5  Rosamund Adoo-Kissi-Debrah    October 2021  422000  12000   
6            Iseult Gillespie   February 2022  412000  12000   
7            Gay Gordon-Byrne     August 2021  455000  13000   
8                 Erin Meezan   February 2022   66000   1900   
9                  Dwan Reece      March 2019  584000  17000   

                                                link  
0  https://ted.com/talks/ozawa_bineshi_albert_cli...  
1  https://ted.com/talks/sydney_iaukea_the_dark_h...  
2  https://ted.com/talks/martin_reeves_how_play_c...  
3  https://ted.com/talks/james_k_thornton_why_is_...  
4  https://ted.com/talks/mahendra_singhi_cement_s...  
5  https://ted.com/talks/rosamund_adoo_kissi_debr...  
6  https://ted.com/talks/iseult_gillespie_the_myt...  
7  https://ted.com/talks/gay_gordon_byrne_you_des...  
8  https://ted.com/talks/erin_meezan_what_nature_...  
9  https://ted.com/talks/dwan_reece_the_origins_o...

Este dataset contiene registros desde 2006 hasta 2022.

In [8]:
talks_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5440 entries, 0 to 5439
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   title   5440 non-null   object
 1   author  5439 non-null   object
 2   date    5440 non-null   object
 3   views   5440 non-null   int64 
 4   likes   5440 non-null   int64 
 5   link    5440 non-null   object
dtypes: int64(2), object(4)
memory usage: 255.1+ KB


Existen 5440 pero tenemos un dato faltante en la columna de Autores.

In [18]:
# Mostramos el registro del autor que falta
talks_df[talks_df['author'].isnull()]

title author           date  views  likes  \
3039  Year In Ideas 2015    NaN  December 2015    532     15   

                                          link  
3039  https://ted.com/talks/year_in_ideas_2015

Esta charla es un resumen, el cual no contiene un autor y se eliminara para la integridad del dataset.

(Te invito a ver el resumen, es intereseante ver como algunas tecnologias evolucionaron rapidamente, pero tambien algunos problemas se mantuvieron) 
